back_ends/r/exec.R

```diff
@@ -28,24 +28,44 @@ if (file.exists(check_code_file_path)) {
   }
 }
 
-exec_jpg <- function(code) {
+exec_jpg <- function(code, i=0) {
   library(ggplot2)
 
   pdf(NULL) # Prevents Rplots.pdf from being created.
 
   eval(parse(text=code))
-  ggsave("/sandbox/image_output", dpi = 150, device = "jpeg")
+  if (i == 0) {
+    ggsave("/sandbox/image_output", dpi = 150, device = "jpeg")
+  }
+  else {
+    ggsave(paste("/sandbox/test_image_output_", i, sep=""), dpi = 150, device = "jpeg")
+  }
 }
 
 code <- readChar(code_file_path, file.info(code_file_path)$size)
 
-if (file.exists(test_code_file_path)) {
-  test_code <- readChar(test_code_file_path, file.info(test_code_file_path)$size)
-  code <- paste(code, test_code, sep="\n\n")
-}
-
 if (output_type == "txt") {
   suppressMessages(suppressWarnings(suppressPackageStartupMessages(eval(parse(text=code)))))
 } else {
   suppressMessages(suppressWarnings(suppressPackageStartupMessages(exec_jpg(code))))
 }
+
+setwd(file.path(tests_dir_path, "outputs"))
+outputs_dir <- getwd()
+
+tests <- list.files(path=tests_dir_path, pattern="test*", full.names=TRUE, recursive=FALSE)
+for (i in seq_along(tests)) {
+    setwd(file.path(outputs_dir, paste("test_", i, sep="")))
+    test_code <- readChar(tests[i], file.info(tests[i])$size)
+
+    # need to find a way to suppress printing to stdout while saving test code output so test outputs don't show up in submission output
```

Comment:
I may be missing it, but do we need to check whether the tests dir exists as we are for the other back ends?